internal/pkg/template/templates/environment/cf.yml

```diff
@@ -69,7 +69,7 @@ Resources:
       Configuration:
         ExecuteCommandConfiguration:
           Logging: DEFAULT
-{{- if .Telemetry }}
+{{- if and .Telemetry .Telemetry.EnableContainerInsights}}
       ClusterSettings:
         - Name: containerInsights
           {{- if .Telemetry.EnableContainerInsights}}
```

Comment:
i think (from https://github.com/golang/go/issues/12995 and some local testing) that this only tests if the `.Telemetry.EnableContainerInsights` is `nil` now that it's a `*bool` - which, because of L72, will always be true. i think you may need to register a "deref" function or something when creating the template to deref the value here